addons/mail/models/mail_message.py

```diff
@@ -721,6 +722,24 @@ def create(self, values):
         if 'record_name' not in values and 'default_record_name' not in self.env.context:
             values['record_name'] = self._get_record_name(values)
 
+        if 'attachment_ids' not in values:
+            values['attachment_ids'] = []
```

Comment:
```python
values.setdefault('attachment_ids', [])
```

And that doesn't seem necessary here, it can just be used at line 738 instead of indexing.